server/gui/main.py

```diff
@@ -237,13 +286,208 @@ def onError(self, err, server_error=False):
             self.serverError = True
             # Report error and switch to load screen
             self.window().shutdownServer()
+        self.resetProgress()
         self.window().stacked_layout.setCurrentIndex(LOAD_INDEX)
         self.error_label.setText(f"Error: {err}")
         self.error_label.resize(self.MAX_CONTENT_WIDTH, self.error_label.height())
+        # self.error_label.setFrameStyle(QFrame.Panel)
 
     onServerError = partialmethod(onError, server_error=True)
 
 
+class AdvancedOptionsArea(QFrame):
+    def __init__(self, parent):
+        super(AdvancedOptionsArea, self).__init__(parent)
+        self.engine_options = {}
+        self.icons = {"collapsed": QIcon(":collapsed.svg"), "expanded": QIcon(":expanded.svg")}
+        self.options_layout = QVBoxLayout()
+        self.header_layout = QHBoxLayout()
+        self.form_layout = QFormLayout()
+        self.form_widget = QFrame()
+        hidden = QSizePolicy()
+        hidden.retainSizeWhenHidden()
+        self.form_widget.setSizePolicy(hidden)
+        self.setupHeader()
+        self.setupForm()
+        self.setLayout(self.options_layout)
+
+    def reset(self):
+        self.title_widget.setText("")
+        self.embedding_widget.setCurrentIndex(0)
+        self.obs_widget.setText("")
+        self.var_widget.setText("")
+        self.max_category_items_widget.setValue(100)
+        self.diff_exp_lfc_cutoff_widget.setValue(0.1)
+        self.form_visibility_toggle.setIcon(self.icons["collapsed"])
+        self.form_widget.hide()
+
+    def setupHeader(self):
+        self.header_layout.addWidget(QLabel("Advanced Options"))
+        self.form_visibility_toggle = QPushButton()
+        self.form_visibility_toggle.setIcon(self.icons["expanded"])
+        self.form_visibility_toggle.clicked.connect(self.toggleOptionsVisibilitiy)
+        self.header_layout.addWidget(self.form_visibility_toggle)
+        self.header_layout.setStretch(0, 4)
+        self.options_layout.addLayout(self.header_layout)
+
+    def setupForm(self):
+        # Option Widgets
+        self.title_widget = QLineEdit()
+        self.title_widget.setAccessibleName("title")
+        self.embedding_widget = QComboBox(self)
+        self.embedding_widget.addItems([""] + MODES)
```

Comment:
why are the layout names hard-coded as a constant?  The scanpy engine derives them from the dataset (see `_default_and_validate_layouts`) as they will vary by dataset.    Suggest that the same happen here.

Reply:
We don't have access to the file at form fill out time. I could do some pre-loading of the file and read from it to get the layouts. I think the point is moot, though, if I go with your CLI options suggestion as planned.

Reply:
in that case, I'd either use my CLI options suggestion, or just remove it entirely...   It is only helpful as a way to reduce the amount of data copied to the front-end....   Otherwise it has little effect....

Reply:
Agreed!